include/TimingTools.h

```diff
@@ -98,6 +98,141 @@ namespace Mezzanine
         /// @param TimingToStream A single NameDuration.
         /// @return The modified stream.
         std::ostream& MEZZ_LIB operator<<(std::ostream& Stream, const NamedDuration& TimingToStream);
+
+        SAVE_WARNING_STATE
+        SUPPRESS_CLANG_WARNING("-Wpadded") // Emscripten complains about this and it not performance sensitive
+
+            /// @brief A set of numbers all the Microbenchmarks return.
+            /// @details This is a collection of numbers intended to provide an ability to get close to deterministic
+            /// results from statistical processes. It is a bad idea to use this to get numbers directly against
+            /// hard-coded values. Rather two things should be benchmarked and something about their relative
+            /// performance should be asserted. One useful test might be that the 90th percentile of the faster item
+            /// should be faster than the 10th percentile of the slower item. This removes much of the impact of
+            /// occasional performance interuptions and does some work producing medians, means and percentiles that
+            /// could be useful in measuring algorithms.
+            /// @n @n
+            /// The closer the performance the less extreme the numbers will diverge. Only in the most extraordinary
+            /// situations will the slowest fast run be faster than the fastest slow run, so compare relative
+            /// percentiles. If 90% of one algorithm is faster than the 10th percentile of another there is a good
+            /// chance it is a keeper. But this can often be a comparison of an algorith with of an algorithm with
+            /// log(N) operations and Linear memory use vs another with Linear CPU use and log(N) memory, and trying to
+            /// infer actual performance on real hardware.
+            struct MEZZ_LIB MicroBenchmarkResults
+            {
+                /// @brief A integral type suitable for counting any reasonable execution counts.
+                using CountType = Mezzanine::UInt64;
+                /// @brief A time precise enough for very small benchmarks.
+                using TimeType = std::chrono::nanoseconds;
+                /// @brief A collection of timings not yet processed
+                using TimingLists = std::vector<TimeType>;
+
+                /// @brief From a collection of nanoseconds fill out this structure.
+                /// @param Timings A list of nanoseconds to measure and get the interesting numbers from.
+                /// @param PrecalculatedTotal Sometimes the total runtime is acquired while running tests. If this is
+                /// non-zero this will be used instead of calculated.
+                MicroBenchmarkResults(const TimingLists& Timings, const TimeType& PrecalculatedTotal = TimeType{0});
+                MicroBenchmarkResults(MicroBenchmarkResults&) = default;
+                ~MicroBenchmarkResults() = default;
```

Comment:
Will doxygen complain about the lack of documentation on these when we get docs up and running?

Reply:
I really don't know what we should do about documenting defaulted members. I am willing to kick that can down the road. Likely we can disable it or document it then, it is unlikely to hold key insights to obscure functionality.